src/Reflection/ClassReflection.php

```diff
@@ -1436,4 +1439,26 @@ public function getResolvedMixinTypes(): array
 		return $types;
 	}
 
+	/**
+	 * @return array<Type>|null
+	 */
+	public function getAllowedSubTypes(): ?array
+	{
+		$subTypes = [];
+
+		foreach ($this->allowedSubTypesClassReflectionExtensions as $allowedSubTypesClassReflectionExtension) {
+			if (!$allowedSubTypesClassReflectionExtension->supports($this)) {
+				continue;
+			}
+
+			$subTypes[] = $allowedSubTypesClassReflectionExtension->getAllowedSubTypes($this);
```

Comment:
do we know a use-case where we would have several extensions for the same type?

I am wondering whether this loop should jus return whatever the first subtype class extensions' `getAllowedSubTypes` returns (which supports the current type)?